faststream/nats/subscriber/factory.py

```diff
@@ -306,3 +318,200 @@ def create_subscriber(
                     description_=description_,
                     include_in_schema=include_in_schema,
                 )
+
+
+def _validate_input_for_misconfigure(
+    subject: str,
+    queue: str,
+    pending_msgs_limit: Optional[int],
+    pending_bytes_limit: Optional[int],
+    max_msgs: int,
+    durable: Optional[str],
+    config: Optional["api.ConsumerConfig"],
+    ordered_consumer: bool,
+    idle_heartbeat: Optional[float],
+    flow_control: Optional[bool],
+    deliver_policy: Optional["api.DeliverPolicy"],
+    headers_only: Optional[bool],
+    pull_sub: Optional["PullSub"],
+    kv_watch: Optional["KvWatch"],
+    obj_watch: Optional["ObjWatch"],
+    ack_first: bool,
+    max_workers: int,
+    stream: Optional["JStream"],
+):
+    if pull_sub is not None and stream is None:
+        raise SetupError("Pull subscriber can be used only with a stream")
+
+    if not subject and not config:
+        raise SetupError("You must provide either `subject` or `config` option.")
+
+    if max_msgs > 0 and any((stream, kv_watch, obj_watch, pull_sub)):
+        warnings.warn(
+            "`max_msgs` option can be used with NATS Core Subscriber - only.",
+            RuntimeWarning,
+            stacklevel=4,
+        )
+
+    if not stream:
+        # Core/Obj/Kv Subscriber
+        if idle_heartbeat is not None:
+            warnings.warn(
+                "`idle_heartbeat` option can be used with JetStream (Pull/Push) - only.",
+                RuntimeWarning,
+                stacklevel=4,
+            )
+
+        if flow_control is not None:
+            warnings.warn(
+                "`flow_control` option can be used with JetStream (Pull/Push) - only.",
+                RuntimeWarning,
+                stacklevel=4,
+            )
```

Comment:
My understanding is that these options only apply to push based consumers.

Reply:
Indeed, fixed